data_and_models/pipelines/ner/preprocess.py

```diff
@@ -0,0 +1,93 @@
+# Adapted from https://github.com/explosion/projects/blob/v3/tutorials/ner_drugs/scripts/preprocess.py.
+
+# MIT License
+#
+# Copyright (c) 2020 ExplosionAI GmbH
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Blue Brain Search is a text mining toolbox focused on scientific use cases.
+#
+# Copyright (C) 2020  Blue Brain Project, EPFL.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+"""
+This script splits and converts annotations from Prodigy <= v1.10.x.
+
+First, this script splits the annotations into train and dev corpora.
+Second, this script converts .jsonl files from Prodigy to .spacy files.
+
+The output is 2 files. Each are named as the input file but with the extension changed.
+The train corpus has the extension ".train.spacy". The dev corpus, ".dev.spacy".
+"""
+
+import typer
```

Comment:
Is this something that comes with `spacy>=3.x` Do we need it?

Reply:
Yes. No.

Reply:
spaCy uses `typer` to power all their CLI commands:
https://github.com/explosion/spaCy/blob/4bd3d01aaf1252d74fda7a94751730bfd83543ab/spacy/cli/project/run.py#L6

I therefore doubt `typer` will be removed from `spaCy` dependencies.

@FrancescoCasalegno Having this new information, would you style recommend a change?

Reply:
Yes I see your point, but I think the same can be said about `dvc`—that library simply cannot work w/o some utility to read YML files so it's quite unlikely that they will drop it.

But I think we should operate in a consistent way: if we decide that _all_ the packages that we `import` in any part of our code must be listed in the `setup.py` and in the `requirements.txt` then this apply also to `typer`.

@Stannislav what do you think?

Reply:
I am open to discussion and would be happy that we define our best practice for this :)

There is actually an issue to discuss how to update `setup.py` and `requirements.txt`.
This is indeed a theme which is coming regularly.
See #316.

Reply:
@FrancescoCasalegno  yes it's absolutely obvious to me that if we explicitely use an external package (e.g. via `import`) then we should also list it as a dependency.

Reply:
OK so then @pafonta can you please add those packages to our dependencies?
Anyway it should not make installation slower, if those packages are also required by `spacy` they won't be installed twice.

Reply:
See also #325.

Reply:
Done in dddd99c.